Src/FluentAssertions/Events/EventAssertions.cs

```diff
@@ -91,18 +91,31 @@ public void NotRaise(string eventName, string because = "", params object[] beca
         public IEventRecording RaisePropertyChangeFor(Expression<Func<T, object>> propertyExpression,
             string because = "", params object[] becauseArgs)
         {
-            IEventRecording recording = monitor.GetRecordingFor(PropertyChangedEventName);
             string propertyName = propertyExpression?.GetPropertyInfo().Name;
 
+            IEventRecording recording = monitor.GetRecordingFor(PropertyChangedEventName);
+
             if (!recording.Any())
             {
                 Execute.Assertion
                     .BecauseOf(because, becauseArgs)
-                    .FailWith("Expected object {0} to raise event {1} for property {2}{reason}, but it did not.",
+                    .FailWith("Expected object {0} to raise event {1} for property {2}{reason}, but it did not raise that event at all.",
                         monitor.Subject, PropertyChangedEventName, propertyName);
             }
 
-            return recording.WithArgs<PropertyChangedEventArgs>(args => args.PropertyName == propertyName);
+            var actualPropertyNames = recording
+                .SelectMany(@event => @event.Parameters.OfType<PropertyChangedEventArgs>())
+                .Select(@args => @args.PropertyName)
+                .Distinct()
```

Comment:
Is there a tests that exercises the distinctiveness?

Reply:
I don’t think changed test exercises the call to `Distinct`. Such a test would need to raise the same event twice and check that the failure message only contains the name of the event once.

Reply:
Adapted the test.